src/box/lua/iproto.c

```diff
@@ -445,6 +447,66 @@ lbox_iproto_send(struct lua_State *L)
 	return (rc == 0) ? 1 : luaT_error(L);
 }
 
+/**
+ * Lua request handler callback: creates new Lua execution context, gets the Lua
+ * callback function, pushes the request header and body as MsgPack objects and
+ * calls the Lua callback.
+ */
+static enum iproto_handler_status
+lua_request_handler_callback(const char *header, const char *header_end,
+			     const char *body, const char *body_end,
+			     void *ctx)
+{
+	struct lua_State *L = luaT_newthread(tarantool_L);
+	if (L == NULL)
+		return IPROTO_HANDLER_ERROR;
+	int coro_ref = luaL_ref(tarantool_L, LUA_REGISTRYINDEX);
+	int callback_ref = (int)(uintptr_t)ctx;
+	lua_rawgeti(L, LUA_REGISTRYINDEX, callback_ref);
+	luamp_push_with_dict(L, header, header_end, iproto_key_translation);
+	luamp_push_with_dict(L, body, body_end, iproto_key_translation);
```

Comment:
AFAIU if someone sends MsgPack with unknown extension, this will raise a Lua error, aborting the server. Please check. We should probably wrap the whole thing in `pcall` if this is true.

Reply:
Acording to tarantool/msgpuck@0c6680a3, no extension checks are currently done.

Reply:
AFAICS we do use extension checking, but it only works for known extensions while unknown extensions are simply [ignored](https://github.com/tarantool/tarantool/blob/7ec8267460a4042001d5232e229795c39cea31f8/src/box/msgpack.c#L109). Anyway, please add a test.

Reply:
You are right, sorry, for some reason my IDE's symbol resolver didn't show [this](https://github.com/tarantool/tarantool/blob/7ec8267460a4042001d5232e229795c39cea31f8/src/box/msgpack.c#L118) place.

Reply:
Discussed with @locker: `luamp_push{_with_dict}` does not raise Lua errors.

Reply:
It does though. Any Lua function can raise them, and `luamp_push()` uses `lua_newuserdata()`. If you open its source, you can see that it calls `lj_err_msg()` sometimes. (Don't get angry again, I will approve whatever.)

Reply:
> Don't get angry again

I didn't get angry... TBH, I enjoyed your previous review, you brought in some really interesting points, especially the one about Lua string hashing. Please excuse me if I sounded toxic or angree, I really didn't mean to be so.

Reply:
> It does though. Any Lua function can raise them, and `luamp_push()` uses `lua_newuserdata()`.

In this case, AFAIC, we assume its okay for the server to abort — I bet I can find other places in our codebase that are not immune to this, e.g.:
https://github.com/tarantool/tarantool/blob/7ec8267460a4042001d5232e229795c39cea31f8/src/box/lua/call.c#L623-L646